Configure

```diff
@@ -14206,6 +14207,82 @@ eval $inlibc
 set getcwd d_getcwd
 eval $inlibc
 
+: check for getenv behavior
```

Comment:
This still performs the test if a value is supplied on the command-line:
```
$ ./Configure -de -Dusedevel -Ud_getenv_preserves_other_thread
....
Checking to see if getenv() preserves a different thread's results
getenv() didn't destroy another thread's buffer
Hmm.  Based on the hints in hints/linux.sh, 
the recommended value for $d_getenv_preserves_other_thread on this machine was "undef"!
    Keep the recommended value? [y]  
...
```

Since this is a relatively expensive test (it builds a program and runs it) it might be good to skip testing if a value is already supplied.  Though @Tux might be the one to ask.

The test itself potentially invokes undefined behaviour, so it might pass at Configure time but the implementation then overwrites in some cases at runtime.

In an ideal world we'd check the documentation of each platform for the equivalent of the "MT-Safe env" from the getenv() documentation for glibc and update the hints files to match, but I don't see us doing that for anything but the major platforms.

It would be good if setting such hints didn't produce a Whoa!

Reply:
This looks like a bug

Reply:
To avoid the message typically you'd check the current value, something like:

```
case "$d_getenv_preserves_other_thread" in
"")
(your probe code here)
;;
esac
```

Reply:
I have added this